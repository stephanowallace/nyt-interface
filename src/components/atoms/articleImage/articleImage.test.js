import React from 'react';
import { shallow } from 'enzyme';
import ContentLoader from 'Components/atoms/contentLoader';
import ArticleImage from '.';

describe('Article image', () => {
  it('renders default state', () => {
    const props = { src: 'test', alt: 'test' };
    const component = shallow(<ArticleImage {...props} />);
    expect(component.find('img').exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('renders loader', () => {
    const component = shallow(<ArticleImage loading />);
    expect(component.contains(<ContentLoader height="75px" width="75px" />)).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('missing src prop', () => {
    const props = { alt: 'test' };
    const component = shallow(<ArticleImage {...props} />);
    expect(component.html()).toBeNull();
  });

  it('missing alt prop', () => {
    const props = { src: 'test' };
    const component = shallow(<ArticleImage {...props} />);
    expect(component.html()).toBeNull();
  });
});
