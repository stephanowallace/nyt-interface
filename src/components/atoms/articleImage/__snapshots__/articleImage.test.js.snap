// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article image renders default state 1`] = `
<figure
  className=""
>
  <img
    alt="test"
    src="test"
  />
</figure>
`;

exports[`Article image renders loader 1`] = `
<div
  className=""
>
  <ContentLoader
    className=""
    height="75px"
    width="75px"
  />
</div>
`;
